### PR TITLE
Tiff: Add support for JPEG quality

### PR DESCRIPTION
**Pull request by olt**
Allow changing the image quality for JPEG compressed TIFFs.

Note: This PR depends on #3861

### CODE REVIEW

**Comment by olt** [comment]
Squashed and rebased on master (#3861). All tests pass.

**Comment by olt** [comment]
Rebased again. All ✅

File: Tests/test_file_libtiff.py [hunk 1]
```diff
@@ -395,18 +435,44 @@ def test_blur(self):
...
+        size_jpeg = os.path.getsize(out)
         im2 = Image.open(out)
         self.assert_image_similar(im, im2, 30)
 
+        im.save(out, compression="jpeg", quality=30)
+        size_jpeg_30 = os.path.getsize(out)
+        im3 = Image.open(out)
+        self.assert_image_similar(im2, im3, 30)
+
+        assert size_raw > size_compressed
```
Comment (hugovk) [inline Tests/test_file_libtiff.py]: Let's use unittest asserts for consistency:
```suggestion
        self.assertGreater(size_raw, size_compressed)
```

File: Tests/test_file_libtiff.py [hunk 2]
```diff
@@ -395,18 +435,44 @@ def test_blur(self):
...
+        assert size_compressed > size_jpeg
```
Comment (hugovk) [inline Tests/test_file_libtiff.py]:
```suggestion
        self.assertGreater(size_compressed, size_jpeg)
```

File: Tests/test_file_libtiff.py [hunk 3]
```diff
@@ -395,18 +435,44 @@ def test_blur(self):
...
+        assert size_jpeg > size_jpeg_30
```
Comment (hugovk) [inline Tests/test_file_libtiff.py]:
```suggestion
        self.assertGreater(size_jpeg, size_jpeg_30)
```

**Comment by olt** [comment]
Merge conflict resolved. Added release notes to #3930

**Comment by hugovk** [comment]
Thank you for your patience! This'll be in 6.1.0.

**Pull request merged**